nl_server/loader.py

```diff
@@ -46,18 +52,66 @@ def load_server_state(app: Any, embeddings_map: Dict[str, str],
     if nl_model and nl_embeddings:
       app.config[config.NL_MODEL_KEY] = nl_model
       app.config[config.NL_EMBEDDINGS_KEY] = nl_embeddings
+      app.config[config.NL_EMBEDDINGS_VERSION_KEY] = embeddings_map
       return
 
   nl_embeddings = embeddings_store.Store(config.load(embeddings_map,
                                                      models_map))
-  app.config[config.NL_EMBEDDINGS_KEY] = nl_embeddings
-
   nl_model = NLAttributeModel()
   app.config[config.NL_MODEL_KEY] = nl_model
+  app.config[config.NL_EMBEDDINGS_KEY] = nl_embeddings
+  app.config[config.NL_EMBEDDINGS_VERSION_KEY] = embeddings_map
 
   if _use_cache(flask_env):
     with Cache(cache.directory, size_limit=_NL_CACHE_SIZE_LIMIT) as reference:
       reference.set(NL_EMBEDDINGS_CACHE_KEY,
                     nl_embeddings,
                     expire=_NL_CACHE_EXPIRE)
       reference.set(NL_MODEL_CACHE_KEY, nl_model, expire=_NL_CACHE_EXPIRE)
+
+
+def _load_yamls(flask_env: str) -> tuple[Dict[str, str], Dict[str, str]]:
+  with open(get_env_path(flask_env, _MODEL_YAML)) as f:
+    models_map = yaml.full_load(f)
+  assert models_map, 'No models.yaml found!'
+
+  with open(get_env_path(flask_env, _EMBEDDINGS_YAML)) as f:
+    embeddings_map = yaml.full_load(f)
+  assert embeddings_map, 'No embeddings.yaml found!'
+
+  custom_map = _maybe_load_custom_dc_yaml()
+  if custom_map:
+    embeddings_map.update(custom_map)
+
+  return embeddings_map, models_map
+
+
+def _maybe_load_custom_dc_yaml():
+  base = os.environ.get('SQL_DATA_PATH')
+  if not base:
+    return None
+
+  file_path = os.path.join(base, f'data/nl/{_CUSTOM_EMBEDDINGS_YAML}')
```

Comment:
Maybe add an optional arg for load_server_state(), which is the folder path, since "data" can become dynamic later.

Reply:
hmm, to confirm, when "data" becomes dynamic, on subsequent boot-up (say server restarts), how will that dynamic path be available?

Reply:
So it almost feels like we should only rely on an env path `CUSTOM_DC_NL_BASE` (which is made of `SQL_DATA_PATH`/data)?

That way, on restart, we can set that env to the new dynamic path.

Reply:
I was thinking the "version" is stored in "SQL_DATA_PATH/version.txt" which can be retrieved, but RSI need to produce that first.

Reply:
That makes sense.   Should we just change loader.py to always read "SQL_DATA_PATH/version.txt" when we have that ready?

So the load flow is same as boot up flow (no override arg for `load/)

Reply:
In any case, added a TODO in this PR for follow up.

Reply:
> I was thinking the "version" is stored in "SQL_DATA_PATH/version.txt" which can be retrieved, but RSI need to produce that first.

That's a good idea.

@shifucun - Can you create a bug for this to record what exactly we want in this file? Also, think about whether a yaml or json might work better for future extensibility.